src/main/java/lex/parser/Parser.java

```diff
@@ -0,0 +1,78 @@
+package lex.parser;
+
+import lex.parser.command.ByeCommand;
+import lex.parser.command.Command;
+import lex.parser.command.DeadlineCommand;
+import lex.parser.command.DeleteCommand;
+import lex.parser.command.EventCommand;
+import lex.parser.command.FindCommand;
+import lex.parser.command.ListCommand;
+import lex.parser.command.MarkCommand;
+import lex.parser.command.TodoCommand;
+import lex.parser.command.UnknownCommand;
+import lex.parser.command.UnmarkCommand;
+import lex.storage.Storage;
+import lex.tasks.TaskList;
+import lex.ui.Ui;
+
+/**
+ * Represents a parser to parse user input.
+ */
+public class Parser {
+    private final TaskList tasks;
+    private final Ui ui;
+    private final Storage storage;
+
+    /**
+     * Constructor for the Parser class.
+     *
+     * @param tasks   The list of tasks.
+     * @param ui      The user interface.
+     * @param storage The storage.
+     */
+    public Parser(TaskList tasks, Ui ui, Storage storage) {
+        this.tasks = tasks;
+        this.ui = ui;
+        this.storage = storage;
+    }
+
+    /**
+     * Parses the user input and returns the corresponding command.
+     *
+     * @param input The user input.
+     * @return The corresponding command.
+     */
+    public Command parse(String input) {
+        String[] inputs = input.split(" ", 2);
+        Instruction instruction;
+
+        try {
+            instruction = Instruction.valueOf(inputs[0].toUpperCase());
+        } catch (IllegalArgumentException e) {
+            return new UnknownCommand(ui);
+        }
+
+        switch (instruction) {
+        case LIST:
```

Comment:
Good job on ensuring that the 'case' is at the same indentation level as the 'switch'. If you haven't already done so (meaning you fixed the indentation manually), you can go to IntelliJ Settings and configure this to be automatic!